modules/table/src/main/java/org/apache/ignite/internal/table/distributed/StorageUpdateHandler.java

```diff
@@ -150,6 +202,25 @@ public void handleUpdateAll(
         });
     }
 
+    private void executeBatchGc(@Nullable HybridTimestamp newLwm) {
+        if (newLwm != null) {
+            HybridTimestamp curLwm = lastRecordedLwm.updateAndGet(prevLwm -> {
+                if (prevLwm == null) {
+                    return newLwm;
+                }
+
+                return newLwm.compareTo(prevLwm) > 0 ? newLwm : prevLwm;
+            });
+
+            if (curLwm == newLwm) {
```

Comment:
Since method `java.util.concurrent.atomic.AtomicReference#updateAndGet` is not single CAS, I think that we will always perform `vacuumBatch`.

Maybe if the new LWM is greater (make a single CAS) or equal to the recorded one, then perform the `vacuumBatch`, otherwise skip.

Reply:
We can't do a single CAS here